core/templates/dev/head/pages/exploration-editor-page/statistics-tab/issues/multiple-incorrect-submissions-issue.directive.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the Multiple Incorrect Submission Issue.
+ */
+
+require('services/exploration-html-formatter.service.ts');
+require('services/html-escaper.service.ts');
+require('domain/utilities/url-interpolation.service.ts');
+require(
+  'pages/exploration-editor-page/statistics-tab/services/' +
+  'learner-action-render.service.ts');
+
+angular.module('oppia').directive('multipleIncorrectSubmissionsIssue', [
+  'ExplorationHtmlFormatterService', 'ExplorationStatesService',
+  'HtmlEscaperService', 'LearnerActionRenderService', 'UrlInterpolationService',
+  'ACTION_TYPE_ANSWER_SUBMIT',
+  function(
+      ExplorationHtmlFormatterService, ExplorationStatesService,
+      HtmlEscaperService, LearnerActionRenderService, UrlInterpolationService,
+      ACTION_TYPE_ANSWER_SUBMIT) {
+    return {
+      restrict: 'E',
+      scope: {},
+      bindToController: {},
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/pages/exploration-editor-page/statistics-tab/issues/' +
+        'multiple-incorrect-submissions-issue.directive.html'),
+      controllerAs: '$ctrl',
+      controller: ['$attrs', function($attrs) {
+        var ctrl = this;
+        ctrl.finalBlock = HtmlEscaperService.escapedJsonToObj(
+          $attrs.finalBlock);
+        ctrl.index = this.finalBlock.length - 1;
+        ctrl.actionStartIndex = $attrs.actionStartIndex;
+        ctrl.stateName =
+          this.finalBlock[this.index].actionCustomizationArgs.state_name.value;
+
+        var _nonSubmitActions = [];
+        for (
+          var i = 0;
+          this.finalBlock[i].actionType !== ACTION_TYPE_ANSWER_SUBMIT; i++) {
+          _nonSubmitActions.push(this.finalBlock[i]);
+        }
+        ctrl.nonSubmitActions = _nonSubmitActions;
+        ctrl.tableIndex = +this.actionStartIndex + i;
```

Comment:
Why the "+" prefix? Perhaps do an explicit cast to be clearer (if that's the intention) when defining ctrl.actionStartIndex above.

Reply:
Done.